Marlin/src/inc/Conditionals_LCD.h

```diff
@@ -615,7 +615,8 @@
 #endif
 
 // Helper macros for extruder and hotend arrays
-#define HOTEND_LOOP() for (int8_t e = 0; e < HOTENDS; e++)
+#define HOTEND_LOOP() for (int8_t e = 0; e < HOTENDS; e++) if (Heater & heater = Temperature::get_heater(HotEndPos0 + e))
+#define HEATER_LOOP() for (uint8_t e = 0; e < HeatersCount; e++) if (Heater & heater = Temperature::get_heater(e))
```

Comment:
These changes are "too smart" and out of scope … and it looks like `&` should be `&&`. If you want a macro like `HOTEND_LOOP` that does more, don't alter `HOTEND_LOOP`. Instead define a new macro in `temperature.h` and replace `HOTEND_LOOP` in the code where needed.

Reply:
Right, that's another magic macro. I needed to iterate on both the hotend index and then get a reference on a current iterated `Heater`, **without** opening a brace since the code does `HOTEND_LOOP() {...}`. The hotend index is accessible in the heater's `pos` member too so I can probably remove the `if( declare variable that always evalutate to true)` trick.

The trick is that the original `HOTEND_LOOP` is not needed anywhere else in the code now.